### PR TITLE
snyk drone plugin

### PR DESCRIPTION
**Pull request by dskatz**
Initial plugin design to run snyk scans.

This is useful for applications that are not supported directly in the Snyk UI (golang) and for adding pipeline steps to fail a build during CI.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+FROM golang:1.13
+
+ENV NODE_VERSION 14
+ENV USER snyk
+ENV USER_ID 999
+ENV USER_HOME /home/${USER}
+
+RUN   groupadd -g ${USER_ID} -r ${USER} \
+  &&  useradd -lmr -u ${USER_ID} -g ${USER} ${USER} \
+  &&  curl -sL https://deb.nodesource.com/setup_${NODE_VERSION}.x | bash - \
```
Comment (redondos) [inline Dockerfile:16]: Don't love this but can live with it. The [manual installation](https://github.com/nodesource/distributions/blob/master/README.md#manual-installation) instructions aren't much better.

File: cmd/drone-snyk/plugin.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+// run the plugin
+func (p *plugin) run() error {
+	err := p.setenv()
+	if err != nil {
+		return err
+	}
+
+	if !checkModeType(p.Mode) {
+		return errors.New("invalid most specified. Only 'test' and 'monitor are valid")
```
Comment (redondos) [inline cmd/drone-snyk/plugin.go:68]:
```suggestion
		return errors.New("invalid mode specified. Only 'test' and 'monitor are valid")
```

File: README.md [hunk 3]
```diff
@@ -1 +1,37 @@
...
+  organization: snyk-organization
+  mode: test
+  auth_token:
+    from_secret: snyk_auth_token
+```
+
+ | Setting | Description |
+ |-------|-----------|
+ | organization | The name of your Snyk Organization |
+ | mode | Supported modes: `test` and `monitor |
```
Comment (redondos) [inline README.md:18]:
```suggestion
 | mode | Supported modes: `test` and `monitor` |
```

File: .drone.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    image: golang:1.13
+    commands:
+      - go vet ./...
+      - go test ./...
+
+  - name: build
+    image: plugins/docker
+    settings:
+      registry: quay.io
+      repo: quay.io/mongodb/dron-snyk
```
Comment (redondos) [inline .drone.yml:17]:
```suggestion
      repo: quay.io/mongodb/drone-snyk
```

File: cmd/drone-snyk/plugin.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	err = cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	// run code scan
+	cmd = exec.Command("snyk", p.getArgs()...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
```
Comment (redondos) [inline cmd/drone-snyk/plugin.go:84]: Am I correct in assuming that `test` will return non-zero and `monitor` will simply report and exit 0?

File: README.md [hunk 6]
```diff
@@ -1 +1,37 @@
...
+Drone plugin for scanning source code using the Snyk CLI.
+
+## Configuration
+
+```yaml
+settings:
+  organization: snyk-organization
+  mode: test
+  auth_token:
+    from_secret: snyk_auth_token
```
Comment (redondos) [inline README.md:12]: Should we look into setting a Drone [org secret](https://docs.drone.io/secret/organization/)?

File: README.md [hunk 7]
```diff
@@ -1 +1,37 @@
 # drone-snyk
+
+Drone plugin for scanning source code using the Snyk CLI.
+
+## Configuration
+
+```yaml
+settings:
+  organization: snyk-organization
```
Comment (redondos) [inline README.md:9]: Are we planning on having multiple Snyk organizations?

File: cmd/drone-snyk/plugin.go (hunk 5 above)
Comment (dskatz) [reply cmd/drone-snyk/plugin.go:84]: Correct.

File: README.md (hunk 6 above)
Comment (dskatz) [reply README.md:12]: Maybe, but most teams have their own organization.

File: cmd/drone-snyk/plugin.go (hunk 5 above)
Comment (dskatz) [reply cmd/drone-snyk/plugin.go:84]: Correct.

File: README.md (hunk 7 above)
Comment (dskatz) [reply README.md:9]: We already do.

File: cmd/drone-snyk/plugin.go (hunk 2 above)
Comment (dskatz) [reply cmd/drone-snyk/plugin.go]: 👍

File: .drone.yml (hunk 4 above)
Comment (dskatz) [reply .drone.yml]: 👍

**Pull request merged**